asllib/ASTUtils.ml

```diff
@@ -593,60 +594,83 @@ let list_cross f li1 li2 =
     [] li1
   |> List.rev
 
+let list_flat_cross f li1 li2 =
+  List.fold_left
+    (fun xys x ->
+      List.fold_left (fun xys' y -> List.rev_append (f x y) xys') xys li2)
+    [] li1
+  |> List.rev
+
```

Comment:
(**
Tests whether the value of binary operation [op] increases along with its left-hand-side operand when all operands are non-negative. That is, if [is_left_increasing op] is true then for every two expressions [x1 op y] and [x2 op y] such that [x1 >= 0 /\ x2 >= 0 /\ y >=0] and [x1 <= x2] then [x1 op y <= x2 op y].
*)